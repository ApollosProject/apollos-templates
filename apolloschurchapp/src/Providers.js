import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <MediaPlayerProvider>
          <AnalyticsProvider>
            <LiveProvider>
              <Providers
                themeInput={customTheme}
                iconInput={customIcons}
                {...props}
              />
            </LiveProvider>
          </AnalyticsProvider>
        </MediaPlayerProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
