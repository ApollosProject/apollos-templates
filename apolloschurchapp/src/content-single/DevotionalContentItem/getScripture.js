import gql from 'graphql-tag';
import ApollosConfig from '@apollosproject/config';

export default gql`
  query getScripture($itemId: ID!) {
    node(id: $itemId) {
      __typename
      id
      ... on DevotionalContentItem {
        scriptures {
          ...ScriptureFragment
        }
      }
    }
  }
  ${ApollosConfig.FRAGMENTS.SCRIPTURE_FRAGMENT}
`;
