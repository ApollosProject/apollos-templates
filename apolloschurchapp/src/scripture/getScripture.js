import gql from 'graphql-tag';
import ApollosConfig from '@apollosproject/config';

export default gql`
  query GetScripture($query: String!) {
    scripture(query: $query) {
      ...ScriptureFragment
    }
  }
  ${ApollosConfig.FRAGMENTS.SCRIPTURE_FRAGMENT}
`;
