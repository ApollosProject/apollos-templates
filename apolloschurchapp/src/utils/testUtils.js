/* eslint-disable import/prefer-default-export */
import renderer from 'react-test-renderer';
import wait from 'waait';

export const renderWithApolloData = async (component) => {
  const tree = renderer.create(component);
  await wait(1);
  tree.update(component);
  return tree;
};
