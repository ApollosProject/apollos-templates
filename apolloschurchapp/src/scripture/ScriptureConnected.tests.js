import React from 'react';

import Providers from '../Providers';

import { renderWithApolloData } from '../utils/testUtils';
import getScripture from './getScripture';
import ScriptureConnected from '.';

const mocks = {
  request: {
    query: getScripture,
    variables: { query: 'Genesis 1:1' },
  },
  result: {
    data: {
      scripture: {
        __typename: 'Scripture',
        id: 'GEN.1.1',
        reference: 'Genesis 1:1',
        copyright: 'PUBLIC DOMAIN',
        version: 'WEB',
        html:
          '<p class="p"><span data-number="1" class="v">1</span>In the beginning, God created the heavens and the earth. </p>',
      },
    },
  },
};

describe('ScriptureConnected component', () => {
  it('renders without errors', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={[mocks]}>
        <ScriptureConnected references={['Genesis 1:1']} />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
});
