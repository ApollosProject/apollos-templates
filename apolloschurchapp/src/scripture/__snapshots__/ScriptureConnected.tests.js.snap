// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptureConnected component renders without errors 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#303030",
        "fontFamily": "InterUI-Bold",
        "fontSize": 16,
        "lineHeight": 24,
        "paddingBottom": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#303030",
          "fontFamily": "InterUI-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        }
      }
    >
      Genesis 1:1
    </Text>
    <Text
      style={
        Object {
          "color": "#505050",
          "fontFamily": "InterUI-Bold",
          "fontSize": 12,
          "lineHeight": 18,
        }
      }
    >
       
      WEB
    </Text>
  </Text>
  <View
    style={
      Object {
        "paddingBottom": 18,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <Text
          bold={false}
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "DroidSerif",
              "fontSize": 16,
              "lineHeight": 26,
            }
          }
        >
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#505050",
                "fontFamily": "DroidSerif",
                "fontSize": 9.6,
                "lineHeight": 26,
              }
            }
          >
             
            <Text>
              1
            </Text>
            Â 
          </Text>
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "DroidSerif",
                "fontSize": 16,
                "lineHeight": 26,
              }
            }
          >
            <Text>
              In the beginning, God created the heavens and the earth. 
            </Text>
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
